modules/services/screen-locker.nix

```diff
@@ -17,81 +29,102 @@ in {
+        Otherwise, this will be used with <command>xset</command> to configure
+        the X server's screensaver timeout.
       '';
     };
 
-    xautolockExtraOptions = mkOption {
-      type = types.listOf types.str;
-      default = [ ];
-      description = ''
-        Extra command-line arguments to pass to <command>xautolock</command>.
-      '';
-    };
-
-    xssLockExtraOptions = mkOption {
-      type = types.listOf types.str;
-      default = [ ];
-      description = ''
-        Extra command-line arguments to pass to <command>xss-lock</command>.
-      '';
-    };
-  };
-
-  config = mkIf cfg.enable {
-    assertions = [
-      (lib.hm.assertions.assertPlatform "services.screen-locker" pkgs
-        lib.platforms.linux)
-    ];
-
-    systemd.user.services.xautolock-session = {
-      Unit = {
-        Description = "xautolock, session locker service";
-        After = [ "graphical-session-pre.target" ];
-        PartOf = [ "graphical-session.target" ];
+    xautolock = {
+      enable = mkOption {
+        type = types.bool;
+        default = true;
+        description = "Use xautolock for time-based locking.";
       };
 
-      Install = { WantedBy = [ "graphical-session.target" ]; };
-
-      Service = {
-        ExecStart = concatStringsSep " " ([
-          "${pkgs.xautolock}/bin/xautolock"
-          "-time ${toString cfg.inactiveInterval}"
-          "-locker '${pkgs.systemd}/bin/loginctl lock-session $XDG_SESSION_ID'"
-        ] ++ optional cfg.enableDetectSleep "-detectsleep"
-          ++ cfg.xautolockExtraOptions);
+      detectSleep = mkOption {
+        type = types.bool;
+        default = true;
+        description = ''
+          Whether to reset xautolock timers when awaking from sleep.
+          No effect if <option>xautolock.enable</option> is false.
+        '';
       };
-    };
 
-    systemd.user.services.xss-lock = {
-      Unit = {
-        Description = "xss-lock, session locker service";
-        After = [ "graphical-session-pre.target" ];
-        PartOf = [ "graphical-session.target" ];
+      extraOptions = mkOption {
+        type = types.listOf types.str;
+        default = [ ];
+        description = ''
+          Extra command-line arguments to pass to <command>xautolock</command>.
+          No effect if <option>xautolock.enable</option> is false.
+        '';
       };
+    };
 
-      Install = { WantedBy = [ "graphical-session.target" ]; };
-
-      Service = {
-        ExecStart = concatStringsSep " "
-          ([ "${pkgs.xss-lock}/bin/xss-lock" "-s \${XDG_SESSION_ID}" ]
-            ++ cfg.xssLockExtraOptions ++ [ "-- ${cfg.lockCmd}" ]);
+    xss-lock = {
+      extraOptions = mkOption {
+        type = types.listOf types.str;
+        default = [ ];
+        description = ''
+          Extra command-line arguments to pass to <command>xss-lock</command>.
+        '';
       };
     };
   };
 
+  config = mkIf cfg.enable (mkMerge [
```

Comment:
@jrobsonchase can you add yourself as a maintainer?

Also, add yourself to `CODEOWNERS` file.